QHyper/converter.py

```diff
@@ -147,60 +148,32 @@ def to_dimod_qubo(problem: Problem, lagrange_multiplier: float = 10
         return cast(tuple[dict[tuple[str, ...], float], float],
                     bqm.to_qubo())  # (qubo, offset)
 
-    # TODO: Refactor is needed
-    @staticmethod
-    def to_dqm(problem: Problem) -> DiscreteQuadraticModel:
-        if hasattr(problem, "one_hot_encoding"):
-            if problem.one_hot_encoding:
-                return Converter._dqm_from_one_hot_representation(problem)
-            return Converter._dqm_from_discrete_representation(problem)
-        return Converter._dqm_from_one_hot_representation(problem)
 
     @staticmethod
-    def _dqm_from_discrete_representation(problem: Problem) -> DiscreteQuadraticModel:
-        dqm = dimod.DiscreteQuadraticModel()
-
-        for var in problem.variables:
-            var = str(var)
-            if var not in dqm.variables:
-                dqm.add_variable(problem.cases, var)
-
-        for vars, bias in problem.objective_function.as_dict().items():
-            x_i, x_j = vars
-            xi_idx: int = cast(int, dqm.variables.index(x_i))
-            xj_idx: int = cast(int, dqm.variables.index(x_j))
-
-            # We're skipping the linear terms
-            if xi_idx == xj_idx:
-                continue
-
-            dqm.set_quadratic(
-                dqm.variables[xi_idx],
-                dqm.variables[xj_idx],
-                {(case, case): bias for case in range(problem.cases)},
-            )
-
-        return dqm
-
-    @staticmethod
-    def _dqm_from_one_hot_representation(problem: Problem) -> DiscreteQuadraticModel:
-        dqm = dimod.DiscreteQuadraticModel()
-
-        CASES_OFFSET = 1 if problem.cases == 1 else 0
+    def to_dqm(problem: Problem) -> DiscreteQuadraticModel:
 
         def binary_to_discrete(v: str) -> str:
             id = int(v[1:])
             discrete_id = id // problem.cases
             return f"x{discrete_id}"
 
-        variables_discrete = [
-            binary_to_discrete(str(v)) for v in problem.variables[:: problem.cases]
-        ]
-        for var in variables_discrete:
-            if var not in dqm.variables:
-                dqm.add_variable(problem.cases + CASES_OFFSET, var)
+        dqm = dimod.DiscreteQuadraticModel()
+
+        objective_function_variables = sorted(problem.objective_function.get_variables(),
+                     key=lambda x: int(x[1:]))
+
+        variables = [
+            binary_to_discrete(str(v)) for v in objective_function_variables[:: problem.cases]
```

Comment:
I don't understand why you are changing ids, could you explain?
And are you sure that `binary_to_discrete` is correct? It will change 'x1', 'x2', ..., 'x9' to 'x0' if `problem.cases == 10`. And that will work only for variables with name as one letter then numbers.

Reply:
Yes, the function is correct.
`dqm.add_variable()` takes the number of cases (in the example you provided, it is 10) and label (it will be 'x0'), so under variable named 'x0' you can input up to ten cases
In CDP example, you can provide to how many communities (to simplyfy subgraphs) you want divide your graph to. If your graph has, for example 10 vertices you end up with 40 variables in problem instance (because of one hot encoding). This functions solves this problem, it provides to DQM 10 variables with 4 cases each